test/helpers/to-have-been-warned.js

```diff
@@ -0,0 +1,44 @@
+if (typeof console === 'undefined') {
+  window.console = {
+    warn: function () {},
+    error: function () {}
+  }
+}
+
+function hasWarned (msg) {
+  var count = console.warn.calls.count()
+  var args
+  while (count--) {
+    args = console.warn.calls.argsFor(count)
+    if (args.some(containsMsg)) {
+      return true
+    }
+  }
+
+  function containsMsg (arg) {
+    if (arg instanceof Error) throw arg
+    return typeof arg === 'string' && arg.indexOf(msg) > -1
+  }
+}
+
+// define custom matcher for warnings
+beforeEach(() => {
+  spyOn(console, 'warn')
+  jasmine.addMatchers({
+    toHaveBeenWarned: () => {
+      return {
+        compare: msg => {
+          var warned = Array.isArray(msg)
+            ? msg.some(hasWarned)
+            : hasWarned(msg)
+          return {
+            pass: warned,
+            message: warned
+              ? 'Expected message "' + msg + '" not to have been warned'
+              : 'Expected message "' + msg + '" to have been warned'
+          }
+        }
+      }
+    }
+  })
+})
```

Comment:
This is copied and modified (console.error => console.warn) from the core repo: https://github.com/vuejs/vue/blob/next/test/helpers/to-have-been-warned.js